src/cli/self_update.rs

```diff
@@ -0,0 +1,276 @@
+#[derive(Debug, clap::Parser)]
+pub struct Args {
+    /// The desired version (to downgrade or upgrade to). Update to the latest version if not specified.
+    #[clap(long)]
+    version: Option<String>,
+
+    /// Force the update even if the pixi binary is not found in the default location.
+    #[clap(long)]
+    force: bool,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Deserialize)]
+struct GithubRelease {
+    tag_name: String,
+    name: String,
+    body: String,
+    html_url: String,
+    assets: Vec<GithubReleaseAsset>,
+}
+
+#[derive(Debug, Deserialize)]
+struct GithubReleaseAsset {
+    name: String,
+    browser_download_url: String,
+}
+
+fn user_agent() -> String {
+    format!("pixi {}", env!("CARGO_PKG_VERSION"))
+}
+
+fn default_archive_name() -> Option<String> {
+    if cfg!(target_os = "macos") {
+        if cfg!(target_arch = "x86_64") {
+            Some("pixi-x86_64-apple-darwin.tar.gz".to_string())
+        } else {
+            Some("pixi-aarch64-apple-darwin.tar.gz".to_string())
+        }
+    } else if cfg!(target_os = "windows") && cfg!(target_arch = "x86_64") {
+        Some("pixi-x86_64-pc-windows-msvc.zip".to_string())
+    } else if cfg!(target_os = "linux") {
+        if cfg!(target_arch = "x86_64") {
+            Some("pixi-x86_64-unknown-linux-musl.tar.gz".to_string())
+        } else if cfg!(target_arch = "aarch64") {
+            Some("pixi-aarch64-unknown-linux-musl.tar.gz".to_string())
+        } else {
+            None
+        }
+    } else {
+        None
+    }
+}
+
+pub async fn execute(args: Args) -> miette::Result<()> {
+    // If args.force is false and pixi is not installed in the default location, stop here.
+    match (args.force, is_pixi_binary_default_location()) {
+        (false, false) => {
+            println!(
+                "{}pixi is not installed in the default location:
+- Default pixi location: {}
+- Pixi location detected: {}
+\nIt can happen when pixi has been installed via a dedicated package manager (such as Homebrew on macOS).
+You can always use `pixi self-update --force` to force the update.",
+                console::style(console::Emoji("✘ ", "")).red(),
+                default_pixi_binary_path().to_str().unwrap(),
+                env::current_exe().unwrap().to_str().unwrap()
+            );
+            return Ok(());
+        }
+        (false, true) => {}
+        (true, _) => {}
+    }
+
+    // Retrieve the target version information from github.
+    let target_version_json = retrieve_target_version(&args.version).await;
+
+    // Get the target version
+    let target_version = target_version_json.tag_name.trim_start_matches('v');
+
+    // Get the current version of the pixi binary
+    let current_version = env!("CARGO_PKG_VERSION");
+    // let current_version = "0.1.0"; // NOTE(hadim): for dev purposes
+
+    // Stop here if the target version is the same as the current version
+    if target_version == current_version {
```

Comment:
The version comparison is basic, but I think it should be fine:

- it's simple (simpler is better xD)
- It allows downgrading the pixi version
- at this point in the code we already know the target version is valid and exists.

Let me know if you prefer something else.